src/Plugin/ServiceDefinition/EntityPost.php

```diff
@@ -32,14 +34,18 @@ class EntityPost extends ServiceDefinitionEntityRequestContentBase {
   public function processRequest(Request $request, RouteMatchInterface $route_match, SerializerInterface $serializer) {
     $entity = parent::processRequest($request, $route_match, $serializer);
     if ($entity) {
-      $entity->save();
-      return [$entity->getEntityType()->id() => $entity->id()];
+      try {
+        $entity->save();
+        if ($entity->id()) {
+          drupal_set_message($this->t("Entity of type @type was created.", ['@type' => $entity->getEntityType()->id()]));
+          return $entity->toArray();
+        }
+      }
+      catch(EntityStorageException $e) {
+        throw new HttpException('500', $e->getMessage());
```

Comment:
What is the best convention here? I saw we were setting response codes and drupal_get_messages.

Are response codes and drupal_set_messages only used for passing states and information? This exception throws a 500 and has it's own messaging